### PR TITLE
hotfix to query WPJ cert using issuers

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Fixed bug related to query of WPJ cert for MAC OS.
*Work in progress. Need to update the same for IOS.

### CODE REVIEW

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 1]
```diff
@@ -51,58 +50,46 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
...
-    if (adError)
-    {
-        if (error)
-        {
-            *error = adError;
-        }
-        MSID_LOG_ERROR(context, @"Failed to retrieve WPJ certificate. Error code: %ld", (long)adError.code);
-        goto _error;
-    }
+    identity = (__bridge_retained SecIdentityRef)[identityDict objectForKey:(__bridge NSString*)kSecValueRef];
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: check that identityDict is not nil

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 2]
```diff
@@ -134,148 +121,25 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
...
-    // specific. The code below searches the identity by passing the WPJ cert as reference.
-    identityQuery = @{ (__bridge id)kSecClass : (__bridge id)kSecClassIdentity,
-                       (__bridge id)kSecReturnRef : (__bridge id)kCFBooleanTrue,
-                       (__bridge id)kSecReturnAttributes : (__bridge id)kCFBooleanTrue,
-                       (__bridge id)kSecAttrKeyClass : (__bridge id)kSecAttrKeyClassPrivate,
-                       (__bridge id)kSecValueRef : (__bridge id)*certificate
-                       };
-    
-    status = SecItemCopyMatching((__bridge CFDictionaryRef)identityQuery, (CFTypeRef*)&result);
+    NSDictionary *query = @{ (__bridge id)kSecClass : (__bridge id)kSecClassIdentity,
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: Does it also return the private key?

**Comment by oldalton** [comment]
Not sure if same fix is necessary also for iOS, as it's querying directly by keychain access group.

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m (hunk 1 above)
Comment (rohitnarula7176) [reply ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: Done

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m (hunk 2 above)
Comment (rohitnarula7176) [reply ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: No it does not. You have to extract the certificate and private key from identity. Previous code created a separate query to get the private key which is not required as you have the identity retrieved already.

**Pull request closed without merging**